java/src/main/java/com/anaconda/skein/ApplicationMaster.java

```diff
@@ -1146,10 +1146,9 @@ public void handleNewContainer(final Container container) {
         // instead. See https://github.com/dask/dask-yarn/issues/48 for more
         // discussion.
         Resource requestedResource = service.getResources();
-        Resource resource = Resource.newInstance(
-            container.getResource().getMemory(),
-            requestedResource.getVirtualCores()
-        );
+        Resource resource = requestedResource;
+        resource.setMemory(container.getResource().getMemory());
```

Comment:
This would need to be a copy for this to work (right now this compares an object with itself).

---

Honestly though, considering how some versions of YARN don't return properly filled-in resource messages, really we could just compare memory here and call it good.

Could you drop the changes here, as well as the creation of the new `resource` object above? Then we can change the safety check below to

```java
if (requestedResource.getMemory() > resource.getMemory()) {
```

This should be good enough - some yarn configurations return resource allocations improperly, but most of the time this safety check should never be hit. Checking for memory alone should be sufficient.

Reply:
ok.. I implemented your proposed change (and yes, I forgot that java doesn't have default copy constructors).